FHEM/00_SIGNALduino.pm

```diff
@@ -4440,76 +4440,55 @@ sub SIGNALduino_querygithubreleases
     HttpUtils_NonblockingGet($param);                                                                                     # Starten der HTTP Abfrage. Es gibt keinen Return-Code. 
 }
 
-sub SIGNALduino_githubParseHttpResponse($)
+sub SIGNALduino_githubParseHttpResponse($$$)
 {
     my ($param, $err, $data) = @_;
     my $hash = $param->{hash};
     my $name = $hash->{NAME};
-
+    my $hardware=AttrVal($name,"hardware",undef);
+    
     if($err ne "")                                                                                                         # wenn ein Fehler bei der HTTP Abfrage aufgetreten ist
     {
         Log3 $name, 3, "error while requesting ".$param->{url}." - $err (command: $param->{command}";                                                  # Eintrag fuers Log
         #readingsSingleUpdate($hash, "fullResponse", "ERROR");                                                              # Readings erzeugen
     }
 
-    elsif($data ne "")                                                                                                     # wenn die Abfrage erfolgreich war ($data enthaelt die Ergebnisdaten des HTTP Aufrufes)
+    elsif($data ne "" && defined($hardware))                                                                                                     # wenn die Abfrage erfolgreich war ($data enthaelt die Ergebnisdaten des HTTP Aufrufes)
     {
     	my $json_array = decode_json($data);
     	#print  Dumper($json_array);
-    	
        	if ($param->{command} eq "queryReleases") {
 	        #Log3 $name, 3, "url ".$param->{url}." returned: $data";                                                            # Eintrag fuers Log
-	
+			
+			my $releaselist="";
 			my @fwreleases;
 			if (ref($json_array) eq "ARRAY") {
 				foreach my $item( @$json_array ) { 
+					next if (AttrVal($name,"updateChannelFW","stable") eq "stable" && $item->{prerelease});
+
 					my %fwrelease;
+					$fwrelease{isprerelease} = $item->{prerelease};
 		
 					#Debug " item = ".Dumper($item);
-					$fwrelease{releasename} = $item->{name};
-					$fwrelease{isprerelease} = $item->{prerelease};
-					$fwrelease{tagname} = $item->{tag_name}; # Anzeige in Auswahlbox
 					
 					foreach my $asset (@{$item->{assets}})
 					{
-				    	my %fileinfo;
-						
-						$fileinfo{filename} = $asset->{name};
-						$fileinfo{dlurl} = $asset->{browser_download_url};
-						$fileinfo{create_date} = $asset->{created_at};
-						push @{$fwrelease{files}}, \%fileinfo;
-						
+						next if ($asset->{name} !~ m/$hardware/i);
+						$releaselist.=$item->{tag_name}."," ;		
```

Comment:
Ich würde es besser finden, wenn in der $releaselist auch das published Datum dabei wäre.
Ich dachte seither, daß das Nightly automatisch jede Nacht erstellt wurde, dem ist aber nicht so.
Ich denke wir können davon ausgehen, daß es einige gibt die regelmässig das Nightly flashen, nur um zu sehen ob es ein neues gibt.
Mit dem Datum hinter dem Tag, ist das flashen auf Verdacht nicht mehr notwendig.  

Z.B. so:
`$releaselist.=$item->{tag_name}."_PubDate=".substr($item->{published_at},0,10)."," ;`

Reply:
Das _PubDate hilft da glaube ich nicht weiter. Das dürfe auf 4. Januar stehen. 

Hinsichtlich des PRs (Ausgewählte Hardware bei der Anzeige der Releases berücksichtigen) sehe ich wenig Zusammenhänge.

Reply:
Es passt aber zu "queryreleases"

Bei einem Nightly Build kann es sein, daß einige davon ausgehen, daß diese Version recht aktuell ist
> Der Nightly Build ist ein Build-Prozess der vornehmlich nachts und automatisiert erfolgt

wenn diese nicht automatisch jede Nacht erstellt wird, ist ein Datum dahinter sinnvoll.

> Das _PubDate hilft da glaube ich nicht weiter.

Passt das create Date evtl besser? wenn nicht, dann ist da z.B. ein compile Date sinnvoll

Reply:
Ich verstehe ehrlich gesagt das Problem nicht und möchte mich hier nicht verzetteln.
Der PR soll die ausgewählte Hardware  bei den gelisteten  Versionen berücksichtigen.

Was würde sich denn ändern, wenn ich das nightly jeden Tag neu compilieren würde? Dann wäre alles gut?

Reply:
> Was würde sich denn ändern, wenn ich das nightly jeden Tag neu compilieren würde? 

Wenn für den Anwender erkennbar ist ob es ein neues Nightly gibt, dann ist das jeden Tag neu compilieren nicht notwendig.
Aktuell finde ich es recht verwirrend, Du bezeichnest eine debug Version die nur ab und zu neu compiliert wird als nightly.
Wie soll der Anwender erkennen ob es eine neue nightly gibt, außer daß er regelmässig neu flasht und schaut ob sich das Datum bei get Version geändert hat.
Mit einem Datum oder einer fortlaufenden Nr könnte es einigen Anwendern viele unnötige Flashvorgänge sparen.
Bei den anderen Releases wäre ein Datum auch interessant, z.B. daß die 3.3.0 von 2016 ist.

Reply:
Soll ich nightly umbenennen?

Das mit dem Datum wird schwierig, es müsste ja das Datum des commits sein und nicht irgend ein anderes.

Reply:
> Soll ich nightly umbenennen?

ja, wäre eine Möglichkeit, dann wäre es weniger verwirrend. Nach der Beschreibung ist es eine Debug Version, hat dies für den Anwender gegenüber der RC10 irgendwelche Nachteile, wenn er sie produktiv einsetzt?

> Das mit dem Datum wird schwierig

Ja, habe ich mittlerweile auch gesehen, daß dies schwierig ist.
Ich sehe 2 Möglichkeiten
- im Tag eine fortlaufende Nr oder Buchstaben
- für jede Version ein neues prerelease, dann müsste das Datum ungefähr passen, dies müsste ausreichend sein

@elektron-bbs @HomeAutoUser 
was meint Ihr dazu?

Reply:
created_at müsste ungefähr passen, bei der nightly ist es 2019-02-03T21:15:30Z

assets: updated_at ist evtl genauer

Reply:
Könnte man nicht einfach das Datum an den Dateinamen anhängen und dieses dann auswerten?

Reply:
wir schweifen hier vom Thema ab. :(